Add project documentation files: README.md, board_rules.md, CHANGELOG.md, release_notes.md. (#25)
ios-course/ironfoudation-team-project: README.md

| ![Xcode](https://img.shields.io/badge/Xcode-14.3-red) | |
| ![Swift](https://img.shields.io/badge/Swift-5.8-orange) | |
| ![SDKs for iOS](https://img.shields.io/badge/SDKs%20for%20iOS-15%2B-lightgrey) | |
| ![SwiftFormat](https://img.shields.io/badge/SwiftFormat-0.51.7-yellow) | [Link](https://github.com/nicklockwood/SwiftFormat) |
| ![SwiftLint](https://img.shields.io/badge/SwiftLint-0.51.0-green)| [Link](https://github.com/realm/SwiftLint) |

## Table of contents
* [General info](#general-info)
* [Installation](#installation)
* [Authors](#authors)

## General info

The repo contains the source code of the `Woof` application.
The `Woof` application is an iOS app that provides the opportunity to connect dog owners and pet sitters to build trusted relationships between them.
You can use the app in 2 modes: like a pet owner or like a pet sitter.
As a pet owner, you can find a sitter for your pet(s) and manage a schedule of your pets' walks, which makes the process of caring for your lovely doggies easier.
As a pet sitter, you can provide the service of walks using your own schedule, prices, and other requirements, if needed.

## Installation

**Pre-requisites**: 

> - installed Xcode with the version specified under the [Current development Environment](https://github.com/ios-course/ironfoudation-team-project/blob/34d602116ed0d8664f4cfb4a11563b4a3dc1f0c2/README.md#current-dev-environment).

Reviewer: You have a link to the specific commit please read the [documentation](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files). Please have a link to the current commit and use anchors as you have done in the table of contents:

```suggestion
- installed Xcode with the version specified under the [Current development Environment](#current-dev-environment).
```

Author: Thanks! My bad. Fixed it.

# WOOF application 

| Dev Environment|
|---------------------|
| ![macOS](https://img.shields.io/badge/macOS-13.3+-blue)|
| ![Xcode](https://img.shields.io/badge/Xcode-14.3-red)  |
| ![Swift](https://img.shields.io/badge/Swift-5.8-orange)|
| ![SDKs for iOS](https://img.shields.io/badge/SDKs%20for%20iOS-15%2B-lightgrey)
| ![SwiftLint](https://img.shields.io/badge/SwiftLint-0.51.0-green): [SwiftLint](https://github.com/realm/SwiftLint)|

> | ![SwiftFormat](https://img.shields.io/badge/SwiftFormat-0.51.7-white): [SwiftFormat](https://github.com/nicklockwood/SwiftFormat)|

Author: Yes, it looks much better). Fixed it.

# WOOF application 

## Current dev environment 

| Dev  environment| Ref link |
| --- | -- |
| ![macOS](https://img.shields.io/badge/macOS-13.3+-blue) | | 
| ![Xcode](https://img.shields.io/badge/Xcode-14.3-red) | |
| ![Swift](https://img.shields.io/badge/Swift-5.8-orange) | |
| ![SDKs for iOS](https://img.shields.io/badge/SDKs%20for%20iOS-15%2B-lightgrey) | |
| ![SwiftFormat](https://img.shields.io/badge/SwiftFormat-0.51.7-yellow) | [Link](https://github.com/nicklockwood/SwiftFormat) |
| ![SwiftLint](https://img.shields.io/badge/SwiftLint-0.51.0-green)| [Link](https://github.com/realm/SwiftLint) |

## Table of contents
* [General info](#general-info)
* [Installation](#installation)
* [Authors](#authors)
* [Notes](#notes)

## General info

This is the application for iOS.

> The `Woof` application provides the opportunity to connect dog owners and pet sitters to build trusted relationships between them. 

Author: Thanks, fixed it.

You can use the app in 2 modes: like a pet owner or like a pet sitter.
As a pet owner, you can find a sitter for your pet(s) and manage a schedule of your pets' walks, which makes the process of caring for your lovely doggies easier.
As a pet sitter, you can provide the service of walks using your own schedule, prices, and other requirements, if needed.

**It also important to know:**

> To process the application some personal information about customers is required, for example, home address, mobile phone.

Author: The paragraph is deleted.

## Installation

**Pre-requisites**: 

> - installed Xcode with the version specified under the Development Environments section.

Author: Fixed it. Added the link to the place in the app, where Development Environment is described.

> Then:

Author: Fixed it.

> Then:

Author: Fixed it.

1) clone the current repository:
[GITHUB Docs](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) 

For example, using the command line:
```bash
$ git clone git@github.com:ios-course/ironfoudation-team-project.git

> ```

Author: Thanks, fixed it.

2) open the file `Woof.xcodeproj` in Xcode.
3) select your device as the destination.
4) press the run button.
If you don't know how to run the app on a real device, please check the [documentation](https://developer.apple.com/documentation/xcode/running-your-app-in-simulator-or-on-a-device)

## Authors

You can contact the authors by the following GitHub's usernames:

- Nadia Shpakouskaya: [@NadzeyaShpakouskaya](https://github.com/NadzeyaShpakouskaya)
- Alex Zavadskiy: [@iBYarchitect](https://github.com/iBYarchitect)
- Julya Kmita: [@julyakmita](https://github.com/julyakmita)

## NOTES:

> More information about working with the application will be added during development.

Author: Fixed it.

# WOOF application 

![macOS](https://img.shields.io/badge/macOS-13.3+-blue) ![Swift](https://img.shields.io/badge/Swift-5.8-orange)  ![Xcode](https://img.shields.io/badge/Xcode-14.3-red) ![SwiftFormat](https://img.shields.io/badge/SwiftFormat-0.51.7-white) ![SwiftLint](https://img.shields.io/badge/SwiftLint-0.51.0-green)

## Table of contents
* [General info](#general-info)
* [Technologies](#technologies)
* [Installation](#installation)
* [Authors](#authors)
* [Notes](#notes)

## General info

This is the application for iOS.
Woof application connects dog owners with pet sitters which contributes to build trusted relationships between them. The app makes it easy for nearby pet owners to find pet sitting and makes convienience of responding to jobs for pet sitters from phone.

> App also makes managing sitting bussines and pet owner's schedule. They can choose their own schedule, prices, and requirements for servise. 

Author: Thank you! )This description is much better. Fixed it.

**It also important to know:**
To process the application some personal information about customers is required, for example, home address, mobile phone.

## Technologies

To contribute code, please, use:
* macOS minimal version - Ventura 13.3
* Xcode version: 14.3
* Swift: 5.8
* SDKs for iOS: 15.0+
* SwiftLint: 0.51.0: [SwiftLint](https://github.com/realm/SwiftLint)
* SwiftFormat: 0.51.7: [SwiftFormat](https://github.com/nicklockwood/SwiftFormat)

> 

Reviewer 2: Maybe make sense to move this section up, because we duplicate our badges and it's easy to forget to update the info in different places.
What do you think?

|  Dev Environment|
| --- | 
 |![macOS](https://img.shields.io/badge/macOS-13.3+-blue) |   |![SwiftLint](https://img.shields.io/badge/SwiftLint-0.51.0-green) | 
| ![Xcode](https://img.shields.io/badge/Xcode-14.3-red) | 
| ![Swift](https://img.shields.io/badge/Swift-5.8-orange) |
| ![SwiftFormat](https://img.shields.io/badge/SwiftFormat-0.51.7-white) | 
| ![SwiftLint](https://img.shields.io/badge/SwiftLint-0.51.0-green)|

Reviewer 3: @NadzeyaShpakouskaya we already have the block with Dev Environment, at the beginning of the README.md

Author: Fixed it.

## Installation

>  Supported devices - iPhone only.

Reviewer 2: Do we need to specify this here?

Author: Removed unnecessary specifying.

 
**Pre-requisites**: 

> To run this project: install Xcode (SDKs for iOS 15.0+ as a deployment target).

Reviewer 2: I think, when we will clone our project, the deployment target will be set, and we don't need to specify it here.
```suggestion
- installed Xcode with the version specified under the Development Environments section.
```

Author: Agree, thanks. Fixed it.

Then:

1) clone the current repository:
[GITHUB Docs](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) 

For example, using the command line:
```bash
$ git clone git@github.com:ios-course/ironfoudation-team-project.git
```

> 2) open the file Woof.xcodeproj in Xcode.

Reviewer 2: nit:
```suggestion
2) open the file `Woof.xcodeproj` in Xcode.
```

Author: Thanks, fixed it.

> 3) select your device as destination.
> [Running your app in Simulator or on a device](https://developer.apple.com/documentation/xcode/running-your-app-in-simulator-or-on-a-device)
> 4) press the run button.

Reviewer 2: nit:
```suggestion
3) select your device as the destination.
4) press the run button.
If you don't know how to run the app on a real device, please check the [documentation](https://developer.apple.com/documentation/xcode/running-your-app-in-simulator-or-on-a-device)
```

Author: Thanks, fixed it.

# WOOF application 

## Table of contents
* [General info](#general-info)
* [Technologies](#technologies)
* [Installation](#installation)
* [Authors](#authors)
* [Notes](#notes)

## General info

This is the application for iOS.
Woof application connects dog owners with pet sitters which contributes to build trusted relationships between them. The app makes it easy for nearby pet owners to find pet sitting and makes convienience of responding to jobs for pet sitters from phone.
App also makes managing sitting bussines and pet owner's schedule. They can choose their own schedule, prices, and requirements for servise. 

**Features include**

> - Secure store all of pet owners and pet sitters databases.

Reviewer: nit: do you have such capabilities now? Are you sure you will implement them at all?

I'd expect this section to explain the current status, not the future plans.

Author: No, we haven't this capabilities now. I'm not  sure that we will implement them all.
The information about feature plans are deleted.

> - Secure store all of pet owners' and pet sitters' data base. 

Reviewer 3: typo:
```suggestion
- Secure store all of pet owners and pet sitters databases. 
```

Author: I wrote in singular form, using possessive syntax. But you are right. It looks now much better. 
Thanks, fixed it.

> - Regularly data base update with personal information.

Reviewer 3: typo:
```suggestion
- Regularly database update with personal information.
```

Author: Thanks, fixed it.

**It also important to know:**
To process the application some personal information about customers is required, for example, home address, mobile phone.

## Technologies

To install the application, please, see the list of requirements and restrictions settings below:

>  * Deployment target for iOS SDKs - 15.0

Reviewer: Do I need Xcode to install app? Is that a requirements to install app?

Reviewer: What version of macOS should I have?

Author: Yes, Xcode is needed. 
macOS Ventura 13.3+

 * Supported devices - iPhone only
 * Supported modes - portrait

>  * Dark/night theme - deactivate

Reviewer: How this info helps me to install the application?

Author: You are right, this information is unnecessary for installation and deleted.

 * The linters (Swiftformat and Swiftlint) integrated with Xcode in the build phase.nters (Swiftformat and Swiftlint) integrated with Xcode in the build phase.

To contribute code, please, use:

> * macOS ventura 13.3

Reviewer: nit:
```suggestion
* macOS Ventura 13.3
```

Author: Thanks, fixed it.

> * macOS ventura 13.3

Author: Minimal version should be 13.3.
Fixed it.

> * macOS ventura 13.3

Author: Also added bages  with environment versions to he beginning of the file.

* Xcode version: 14.3
* Swift: 5.8
* SDKs for iOS: 15.0
* SwiftLint: 0.51.0

> * SwiftFormat: 0.51.7

Reviewer: nit: it's good to give a link to a 3rd-parry tools to vaoid ambiguities.

Author: Thanks, the links are added.

## Installation

**Pre-requisites**: 
To run this project: install Xcode (SDKs for iOS 15.0 as a deployment target).
Then:

> 1) clone the current repository via ssh: 

Reviewer: So I can't clone it via HTTPS?

Author: My bad, it was not clearly description. Of course, you can.  Fixed it.

> ```

Author: Thanks, fixed it.

> git@github.com:ios-course/ironfoudation-team-project.git

Reviewer: I've tried to run this command and got an error. What should I do?

Author: Sorry for this. Fixed it.

```

> 2) open the file Woof.xcodeproj in Xcode

Reviewer: OK, I open the `Woof.xcodeproj` file and then what?
I still didn't install application. What should I do?

- Regularly data base update with personal information.

**It also important to know:**
To process the application some personal information about customers is required, for example, home address, mobile phone.

## Technologies

To install the application, please, see the list of requirements and restrictions settings below:

```
 Deployment target for iOS SDKs - 15.0
 Supported devices - iPhone only
 Supported modes - portrait
 Dark/night theme - deactivate
 The linters (Swiftformat and Swiftlint) integrated with Xcode in the build phase.
```
To contribute code, please, use:

* macOS ventura 13.3
* Xcode version: 14.3
* Swift: 5.8
* SDKs for iOS: 15.0
* SwiftLint: 0.51.0
* SwiftFormat: 0.51.7

## Installation

**Pre-requisites**: 
To run this project: install Xcode (SDKs for iOS 15.0 as a deployment target).
Then:

1) clone the current repository via ssh: 

```
git@github.com:ios-course/ironfoudation-team-project.git
```

2) open the file Woof.xcodeproj in Xcode

## Authors

You can contact the authors by the following GitHub's usernames:

```
Nadia Shpakouskaya: @NadzeyaShpakouskaya 
Alex Zavadskiy: @iBYarchitect
Julya Kmita: @julyakmita

> ```

Author: Yes, much better! Thanks, fixed it.

> # ironfoudation-team-project

Author: Thanks, fixed it.

# WOOF application 

## Table of contents
* [General info](#general-info)
* [Technologies](#technologies)
* [Installation](#installation)
* [Sources](#sources)
* [Authors](#authors)
* [Important](#important)
* [Notes](#notes)

## General info

This is the application for iOS.
Woof application connects dog owners with pet sitters which contributes to build trusted relationships between them. The app makes it easy for nearby pet owners to find pet sitting and makes convienience of responding to jobs for pet sitters from phone.
App also makes managing sitting bussines and pet owner's schedule. They can choose their own schedule, prices, and requirements for servise. 

**Features include**
-Secure store all of pet owners' and pet sitters' data base. 
-Regularly data base update with personal information.

> 

Author: Thanks, spaces were added.

**It also important to know:**

> To process application,some private cliets information like home adress, mobile phone is needed.

Reviewer 2: Please, check the typos.

```suggestion
To process the application some personal information about customers is required, for example, home address, mobile phone.
```

Author: Thanks, the typos are fixed.

## Technologies

> Project is created with:

Author: The restrictions are added.

* macOS ventura 13.3
* Xcode version: 14.3
* Swift: 5.8

> * SDKs for IOS: 16.4

Author: Sorry for this typo! Thanks, fixed it.

> * SDKs for IOS: 16.4

Author: Yes, my mistake. Thanks, fixed it.

> * SwiftLint: 0.51.0

Author: @dersim-davaod I need help with this task. I didn't badges using shields.io:
'https://shields.io/category'
that are connected with iOS, Swift etc.

* SwiftFormat: 0.51.7

## Installation

> To run this project:

Reviewer: Before this: are there any restrictions/pre-requisites? For example, should I install Xcode?

Author: Thanks for the comment, restrictions/pre-requisites are added.

```

> 1)install it locally by cloning the current repository, using: git@github.com:ios-course/ironfoudation-team-project.git

Author: Thanks, fixed it.

> 2)run the file Woof.xcodeproj

Reviewer: ```suggestion
2) open the file Woof.xcodeproj in Xcode
```

Author: Thanks, fixed it.

> ```

Reviewer 2: Maybe better to format the code like this, it's hard to read  this file:

To run this project:
1) install it locally by cloning the current repository via ssh:
```
 git@github.com:ios-course/ironfoudation-team-project.git
```
2) run the file Woof.xcodeproj

Author: LGTM, thanks for a good advice. Updated format.

## Sources

> This app uses the folowing API and libraries:

Reviewer: I'd suggest to remove this section as we don't use any API at the moment.

Author: Thanks, fixed it.

//*This section will be updated during the application development*  

## Authors

You can contact the authors by the following email adresses:

> ```
> Nadia Shpakouskaya: shpakovna@gmail.com 
> Alex Zavadskiy: fh.zavadskii@gmail.com
> Julya Kmita: julyakmita@gmail.com
> ```

Reviewer 2: Please remove the emails, I don't want to share this information, you can use github links to our pages instead.
```suggestion
```
Nadia Shpakouskaya: @NadzeyaShpakouskaya 
Alex Zavadskiy: @iBYarchitect 
Julya Kmita: @julyakmita 
```
```

Reviewer: Agree! This is already shared unfortunately, as the repo as public.

Reviewer: I recommend to remove the commit and force push changes to remove the changes from history also.

Author: Sorry for this sharing of private information! The commit is deleted (the changes from history also).  The emails are replaced by nicknames on GitHub.

## Important:

### Creating a new branch:

> 1) all brunches are created from the main branch;

Author: Thanks, fixed it. This information has been moved to the file  _board_rules.md_.

2) name of a new branch should show the issue type, issue number and branch name:
 ```
 [type_of_task - #issue_from_backlog - title].
 For example:
 feature-1-create-project.
 ```
### Merge restrictions to the main branch:
-minimum two approvals;

> -all conversations should be closed (resolve) after exhaustive discussion.

Reviewer 2: Maybe better move this information to board_rules or another place.
I don't think, that this information is important for users.

Author: I agree. The information is removed to the board_rules file.

## Technologies

To install the application, please, see the list of requirements and restrictions settings below:

```
 Deployment target for iOS SDKs - 15.0
 Supported devices - iPhone only
 Supported modes - portrait
 Dark/night theme - deactivate
 The linters (Swiftformat and Swiftlint) integrated with Xcode in the build phase.

> ```

Author: For me both cases are readable. I followed your suggestion and reformatted this lines.
